great_expectations/rule_based_profiler/profiler.py

```diff
@@ -108,26 +116,40 @@ def __init__(
         These will be used to define profiler computation patterns.
 
         Args:
-            profiler_config: Variables and Rules configuration as a dictionary
+            name: The name of the RBP instance
+            config_version: The version of the RBP (currently only 1.0 is supported)
+            rules: A set of RuleConfigs, each of which contains its own DomainBuilder, ParameterBuilders, and ExpectationConfigurationBuilders
+            variables: Any variables to be substituted within the rules
             data_context: DataContext object that defines a full runtime environment (data access, etc.)
         """
-        self._profiler_config = profiler_config
+        self._name = name
+        self._config_version = config_version
         self._data_context = data_context
+        self._variables = variables or {}
         self._rules = []
 
-        rules_configs: Dict[str, Dict[str, Any]] = self._profiler_config.get(
-            "rules", {}
-        )
-        rule_name: str
-        rule_config: Dict[str, Any]
-
-        for rule_name, rule_config in rules_configs.items():
-            domain_builder_config: dict = rule_config.get("domain_builder")
+        # Necessary for citation
+        self._profiler_config = {
+            "name": name,
+            "config_version": config_version,
+            "rules": rules,
+            "variables": variables,
+        }
```

Comment:
This is a weird one - we need the raw rules and variables when creating a citation but want the converted values for purposes of the constructor.

Perhaps there's a nicer way to do this but proposing the change that has the least impact on current behavior as this will all probably be refactored shortly